build/bazelutil/stamp.sh

```diff
@@ -66,7 +66,12 @@ else
     shift 1
 fi
 
-BUILD_REV=$(git describe --match="" --always --abbrev=40 --dirty)
+if [[ "$(git status -s --ignore-submodules --untracked-files=no)" ]]; then
+  BUILD_REV=$(git describe --match="" --always --abbrev=40 --dirty)
```

Comment:
Remember the point of this whole thing is to avoid using the `--dirty` flag to `git describe`, so if you do it anyway there is no point to the change. The point of my explanation (in DM) is that we want to check whether `git status -s --ignore-submodules --untracked-files=no` reports any changes, and if so, then append the `-dirty` suffix ourselves.

That would look like this:

```
BUILD_REV=$(git describe --match="" --always --abbrev=40)
if [[ ! -z "$(git status -s --ignore-submodules --untracked-files=no)" ]]; then
    BUILD_REV=$BUILD_REV-dirty
fi
```

Reply:
Yeah it looks like your code just does the same thing that mine does, except a little neater. Will change.

Reply:
Clarified in DM, but no, the two code snippets are functionally different.